src/System.Private.CoreLib/shared/Interop/Unix/System.Globalization.Native/Interop.Collation.cs

```diff
@@ -27,6 +27,8 @@ internal static partial class Globalization
 
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_LastIndexOf")]
         internal static extern unsafe int LastIndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options);
+        [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_LastIndexOf")]
+        internal static extern unsafe int LastIndexOf(SafeSortHandle sortHandle, char* target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options);
```

Comment:
Do we need the existing LastIndexOf overload once this new one is added?  It looks like there's only one call site to the existing one, and it could use the new one simply by adding a `fixed (char* pTarget = target)` line:
https://github.com/dotnet/coreclr/blob/bcabe118b783feb489c791e340f1c3ea792acde2/src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs#L451-L453

Reply:
The code path you are talking about is when using LastIndexOf on strings on Unix.
Don't you think that pinning the string does regress perf slightly? If you think we should address it, we should also do the same for IndexOf.

Reply:
> Don't you think that pinning the string does regress perf slightly? 

That pinning has to happen regardless. This would just move it from one place (PInvoke stub) to different place. As always, you can measure to double check.

Reply:
I see. Thanks, changed them for both IndexOf and LastIndexOf.